pkg/metadata/host/host.go

```diff
@@ -189,7 +207,16 @@ func getNetworkMeta() *NetworkMeta {
 		log.Infof("could not get network metadata: %s", err)
 		return nil
 	}
-	return &NetworkMeta{ID: nid}
+
+	networkMeta := &NetworkMeta{ID: nid}
+
+	publicIPv4, err := getPublicIPv4()
+	if err == nil {
+		log.Infof("Adding public IPv4 %s to network metadata", publicIPv4)
+		networkMeta.PublicIPv4 = publicIPv4
+	}
```

Comment:
Is it ok that we are completely ignoring the `err` here? On failure, nothing would be logged, which could make it hard to debug issues. Perhaps some debug/trace level logs in `getPublicIPv4()` would be appropriate.